extensions/amp-date-display/1.0/amp-date-display.js

```diff
@@ -76,14 +45,50 @@ class AmpDateDisplay extends PreactBaseElement {
     );
     return isLayoutSizeDefined(layout);
   }
+
+  /** @override */
+  checkPropsPostMutations() {
+    const templates =
+      this.templates_ || (this.templates_ = Services.templatesFor(this.win));
+    const template = templates.maybeFindTemplate(this.element);
+    if (template != this.template_) {
+      this.template_ = template;
+      if (template) {
+        // Only overwrite `render` when template is ready to minimize FOUC.
+        templates.whenReady(template).then(() => {
+          if (template != this.template_) {
+            // A new template has been set while the old one was initializing.
+            return;
+          }
+          this.mutateProps(
+            dict({
+              'render': (data) => {
+                return templates
+                  .renderTemplateAsString(dev().assertElement(template), data)
```

Comment:
Nit: Maybe we should be extracting this out of the render function to the outer. That way, we don't mutate the `render` prop if the template has changed between the `whenReady` and the `renderTemplateAsString`.

Reply:
Could you pls clarify? This currently waits until the template has been loaded between updating the render function. That way, the `render` is only set when it can actually be used.

Reply:
Ohh, I failed to see that `data` is needed for the render. I was going to suggest that we do:

```js
renderTemplateAsString(el).then(html => {
  this.mutateProps({
    render() {
      return { __html: html };
    }
  }
});
```

But that won't work because we need `data`.

Reply:
Yeah. I tried variants more similar to this as well. But I think the latest is a good compromise.